### ISSUE TITLE
Incorrect type annotation on Multimetric.compute()

### ISSUE THREAD

**Issue by gabbard** [google/flax#4481]
[The type signature of `Multimetric.compute`](https://github.com/google/flax/blob/1961c1259e6c2d22a0c27de9599476aaa073a553/flax/nnx/training/metrics.py#L376) seems to be incorrect. It is currently:

```python
def compute(self) -> dict[str, Metric]:
```

Its implementation is

```
return {
        f'{metric_name}': getattr(self, metric_name).compute()
        for metric_name in self._metric_names
    }
```

meaning that the second type parameter of the return type should be the return type of `Metric.compute()`.  [`Metric.compute` itself does not specify a return type, or even if it returns anything](https://github.com/google/flax/blob/1961c1259e6c2d22a0c27de9599476aaa073a553/flax/nnx/training/metrics.py#L50).  The provided implementations (besides `Multimetric` itself!) return `jax.Array` and `nnx.training.metrics.Statistics` objects, suggesting (reasonably) that no restriction is placed on the return.  

It looks like:
* the return type of `Metric.compute` should be either `Any` or `Optional[Any]`  depending on whether `Metric.compute()` has to return something (I would assume it does?).
* the return type of `MultiMetric.compute` should be `dict[str, Any]` or `dict[str, Optional[Any])`.

### System information
- Flax, jax, jaxlib versions (obtain with `pip show flax jax jaxlib`: 0.10.2

**Comment by cgarciae** [comment]
Thanks @gabbard, I think it should be `dict[str, Any]`. Created #4485 to fix this.
